yo/app/scripts/controllers/datasetsCtrl.js

```diff
@@ -36,96 +44,162 @@ angular.module('hopsWorksApp')
+                        self.currentBoard = JSON.parse(response.board);
+                        self.initializeMetadataTabs(JSON.parse(response.board));
+
+                        self.fetchMetadataForTemplate();
+                      });
+            };
+
+            self.fetchMetadataForTemplate = function () {
+              //columns are the tables in the template
+              self.meta = [];
+
+              var tables = self.currentBoard.columns;
+              angular.forEach(tables, function (table, key) {
+                //console.log("value " + JSON.stringify(table));
+                MetadataActionService.fetchMetadata(table.id, self.currentFile.id)
+                        .then(function (response) {
+                          //console.log("METADATA FOR TABLE " + table.name);
+                          //console.log("ARE " + JSON.stringify(response.board));
+                          self.reconstructMetadata(table.name, JSON.parse(response.board));
+                          //self.meta = JSON.parse(response.board);
+                        });
+              });
+            };
+
+            self.reconstructMetadata = function (tableName, rawdata) {
+
+              $scope.tableName = rawdata.table;
+//                    self.metadataView = results;
+//                    self.meta.push({name: tableName, tuples: self.metadataView});
+
+              self.meta.push({name: tableName, rest: rawdata});
+              self.metadataView = {};
+              console.log(JSON.stringify(self.meta));
+              //self.print(self.meta);
+              console.log("RECONSTRUCTED ARRAY  " + JSON.stringify(self.meta));
+            };
+
+            self.print = function (meta) {
+              angular.forEach(meta, function (table) {
+                var field = table.tuples.headers;
+                console.log("IN TABLE " + table.name);
+
+                angular.forEach(field, function (fieldName) {
+
+                  console.log("IN FIELD " + fieldName);
+
+                  var valueArray = table.tuples.values;
+
+                  angular.forEach(valueArray, function (value) {
+                    console.log("printing " + JSON.stringify(value[0].raw));
+                  });
+                });
+              });
+            };
+
+            self.initializeMetadataTabs = function () {
+              self.tabs = [];
+
+              angular.forEach(self.currentBoard.columns, function (value, key) {
+                console.log(key + ': ' + value.name);
+                self.tabs.push({title: value.name, cards: value.cards});
               });
+              //console.log("initialized tabs " + JSON.stringify(self.tabs));
             };
 
-            /**
-             * Download a file.
-             * @param {type} file Can either be a filename or a path.
-             * @returns {undefined}
+            /*
```

Comment:
You should move functionality concerning only metadata to a separate controller.